frontend/components/LSP7Panel.tsx

```diff
@@ -93,10 +94,13 @@ const LSP7Panel: React.FC<LSP7PanelProps> = ({
         signer
       );
       const lsp7 = tokenContract.connect(signer);
+    
+      const balanceInBaseUnits = ethers.utils.parseUnits(tokenData?.balance as string, tokenData?.decimals);
+
       const lsp7Tx = lsp7.interface.encodeFunctionData('transfer', [
         vaultAddress,
         await signer.getAddress(),
-        tokenData?.balance,
+        balanceInBaseUnits,
```

Comment:
this could be wrong value if you dont convert data when we first get it

Reply:
the conversion is in line 98